multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/jobs/OrphanedFilesCleaner.java

```diff
@@ -35,69 +37,77 @@ public class OrphanedFilesCleaner {
     private final ApplicationConfiguration configuration;
 
     @Inject
-    public OrphanedFilesCleaner(FileService fileService, OperationService operationService,
-                                FlowableFacade flowableFacade, ApplicationConfiguration config) {
+    public OrphanedFilesCleaner(FileService fileService, OperationService operationService, FlowableFacade flowableFacade,
+                                ApplicationConfiguration config) {
         this.fileService = fileService;
         this.operationService = operationService;
         this.flowableFacade = flowableFacade;
         this.configuration = config;
     }
 
-    //this is quite inefficient because the Operation table does not contain any info on which archive
-    //it is started with
+    // this is quite inefficient because the Operation table does not contain any info on which archive
+    // it is started with
     @Scheduled(fixedRate = 30, timeUnit = TimeUnit.MINUTES)
     public void run() {
         if (configuration.getApplicationInstanceIndex() != SELECTED_INSTANCE_FOR_CLEANUP) {
             return;
         }
 
         var timestamp = LocalDateTime.now();
-        var oneHourAgo = timestamp.minusHours(1);
+        var oneAndHalfHoursAgo = timestamp.minusHours(1)
+                                          .minusMinutes(30);
+        var halfHourAgo = timestamp.minusMinutes(30);
         try {
-            LOGGER.debug(MessageFormat.format(Messages.GETTING_FILES_CREATED_AFTER_0, oneHourAgo));
-            var files = fileService.listFilesCreatedAfter(oneHourAgo);
+            LOGGER.info(MessageFormat.format(Messages.GETTING_FILES_CREATED_AFTER_0_AND_BEFORE_1, oneAndHalfHoursAgo, halfHourAgo));
+            var files = fileService.listFilesCreatedAfterAndBefore(oneAndHalfHoursAgo, halfHourAgo);
             var fileIdsToFiles = files.stream()
-                                      .collect(Collectors.toMap(FileEntry::getId, entry -> entry));
-
-            var historicVariables = getHistoricAppArchiveIDs(oneHourAgo);
+                                      .collect(Collectors.toMap(FileEntry::getId, Function.identity()));
 
-            filterFilesWithStartedOperations(fileIdsToFiles, historicVariables);
+            var archiveIdsWithAssociatedOperation = getHistoricAppArchiveIDs(oneAndHalfHoursAgo, halfHourAgo);
```

Comment:
Why do we need to make this list more specific - wouldn't this mean that operations started in the last half hour wouldn't get filtered out and corresponding files will be marked for deletions.

Reply:
the operations and files added in the last half an hour will not be added to this list. We need to filter out the files which are associated with an operation because fileIdsToFiles initially contains all the files for the 1 hour timeframe